apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4SamhsaMatcher.java

```diff
@@ -522,43 +524,95 @@ private static String normalizeIcd10Code(String icd10DiagnosisCode) {
   /**
    * @param procedureConcept the procedure {@link CodeableConcept} to check
    * @return <code>true</code> if the specified procedure {@link CodeableConcept} contains any
-   *     {@link Coding}s that match any of the {@link #cptCodes}, <code>false</code> if they all do
-   *     not
+   *     {@link Coding}s that match any of the {@link #cptCodes} or has unknown coding systems,
+   *     <code>false</code> otherwise
    */
   private boolean containsSamhsaProcedureCode(CodeableConcept procedureConcept) {
+    // If there are no procedure codes, then we cannot have any blacklisted codes
+    if (procedureConcept.getCoding().isEmpty()) {
+      return false;
+    }
+
+    // Does the CodeableConcept have a legit HCPCS Coding?
+    boolean hasHcpcsCoding = hasHcpcsCoding(procedureConcept);
+
+    // Check that Coding to see if it's blacklisted.
+    if (hasHcpcsCoding && isSamhsaCptCode(procedureConcept)) {
+      return true;
+    } else if (!containsOnlyKnownSystems(procedureConcept)) {
```

Comment:
The only thing I'm not thrilled about with this is that it technically iterates over the same codings 3 separate times (assuming you made it through the `else if` block)

Reply:
Good point, this can be simplified at least by merging the first two iterations into one and making a single method for the first condition.